geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/serialization/WkbSerialization.scala

```diff
@@ -191,4 +194,15 @@ object WkbSerialization {
   val Point: Int              = 8
   val LineString: Int         = 9
   val Polygon: Int            = 10
+
+  private def is2d(geometry: Geometry): Boolean = {
+    // don't trust coord.getDimensions - it always returns 3 in jts
+    // instead, check for NaN for the z dimension
+    // note that we only check the first coordinate - if a geometry is written with different
+    // dimensions in each coordinate, some information may be lost
+    val coord = geometry.getCoordinate
+    // check for dimensions - use NaN != NaN to verify z coordinate
+    // TODO check for M coordinate when added to JTS
```

Comment:
JTS already supports measure dimensions somewhat.  Folks would likely have to go to lengths to work around some deficiencies in the WKTReader/WKBReader.

Since this TODO would be a decent size change, it is worth entering a JIRA ticket to the tune of "Support M dimensions"?  (This would give us a clear place to signal that GeoMesa doesn't do it, and give a place to discuss it.)

Reply:
this TODO was copied over from the original check. How does jts support M, since all coords have 3 dimensions? I think it would be pretty easy for us to implement if jts supports it.

Reply:
https://github.com/locationtech/jts/blob/jts-1.16.0/modules/core/src/test/java/org/locationtech/jts/io/WKTReaderTest.java#L90

https://github.com/locationtech/jts/blob/master/modules/core/src/test/java/test/jts/GeometryTestCase.java#L178-L201

Those might help give some ideas.

Reply:
nice, looks like it was added in 1.16. guess we should read release notes :P
opened https://geomesa.atlassian.net/browse/GEOMESA-2877

Reply:
Yeah, are you talking about the release notes that I evidently wrote here: https://github.com/locationtech/jts/releases/tag/jts-1.16.0 ?  That's why I had a vague recollection of JTS supporting ZM;).

Reply:
:joy: